libraries/standard/mqtt/integration-test/mqtt_system_utest.c

```diff
@@ -0,0 +1,700 @@
+                                  SSL * pSslContext,
+                                  bool createCleanSession,
+                                  bool * pSessionPresent );
+
+/**
+ * @brief The application callback function that is expected to be invoked by the
+ * MQTT library for incoming publish and incoming acks received over the network.
+ *
+ * @param[in] pContext MQTT context pointer.
+ * @param[in] pPacketInfo Packet Info pointer for the incoming packet.
+ * @param[in] packetIdentifier Packet identifier of the incoming packet.
+ * @param[in] pPublishInfo Deserialized publish info pointer for the incoming
+ * packet.
+ */
+static void eventCallback( MQTTContext_t * pContext,
+                           MQTTPacketInfo_t * pPacketInfo,
+                           uint16_t packetIdentifier,
+                           MQTTPublishInfo_t * pPublishInfo );
+
+/*-----------------------------------------------------------*/
+
+static void establishMqttSession( MQTTContext_t * pContext,
+                                  SSL * pSslContext,
+                                  bool createCleanSession,
+                                  bool * pSessionPresent )
+{
+    MQTTConnectInfo_t connectInfo;
+    MQTTTransportInterface_t transport;
+    MQTTFixedBuffer_t networkBuffer;
+    MQTTApplicationCallbacks_t callbacks;
+
+    assert( pContext != NULL );
+    assert( pSslContext != NULL );
+
+    /* The network buffer must remain valid for the lifetime of the MQTT context. */
+    static uint8_t buffer[ NETWORK_BUFFER_SIZE ];
+
+    /* Setup the transport interface object for the library. */
+    transport.networkContext = pSslContext;
+    transport.send = transportSend;
+    transport.recv = transportRecv;
+
+    /* Fill the values for network buffer. */
+    networkBuffer.pBuffer = buffer;
+    networkBuffer.size = NETWORK_BUFFER_SIZE;
+
+    /* Application callbacks for receiving incoming publishes and incoming acks
+     * from MQTT library. */
+    callbacks.appCallback = eventCallback;
+
+    /* Application callback for getting the time for MQTT library. This time
+     * function will be used to calculate intervals in MQTT library.*/
+    callbacks.getTime = getTimeMs;
+
+    /* Initialize MQTT library. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_Init( pContext,
+                                               &transport,
+                                               &callbacks,
+                                               &networkBuffer ) );
+
+    /* Establish MQTT session with a CONNECT packet. */
+
+    connectInfo.cleanSession = createCleanSession;
+
+    connectInfo.pClientIdentifier = TEST_CLIENT_IDENTIFIER;
+    connectInfo.clientIdentifierLength = TEST_CLIENT_IDENTIFIER_LENGTH;
+
+    /* The interval at which an MQTT PINGREQ needs to be sent out to broker. */
+    connectInfo.keepAliveSeconds = MQTT_KEEP_ALIVE_INTERVAL_SECONDS;
+
+    /* Username and password for authentication. Not used in this demo. */
```

Comment:
Not used in this demo. -->Not used in this test.

Reply:
Fixed